src/data_factory_testing_framework/_expression_runtime/data_factory_expression/expression_transformer.py

```diff
@@ -45,38 +39,37 @@ def __init__(self) -> None:
         """
 
         expression_grammar = f"""
-            // TODO: add support for array index
-            ?expression_start: expression_evaluation
-            # TODO: probably object accessor does not apply to all below in expression_evaluation
+            expression_start: "@" expression_evaluation
             expression_evaluation: (expression_logical_bool | expression_branch | expression_call) ((("." EXPRESSION_PARAMETER_NAME) | EXPRESSION_ARRAY_INDEX)+)?
             ?expression_call: expression_function_call
-                                    | expression_pipeline_reference
-                                    | expression_variable_reference
-                                    | expression_activity_reference
-                                    | expression_dataset_reference
-                                    | expression_linked_service_reference
-                                    | expression_item_reference
-                                    | expression_system_variable_reference
+                              | expression_variable_reference
+                              | expression_item_reference
+                              | expression_datafactory_parameters_reference
+                              | expression_datafactory_activity_reference
+                              | expression_parameter_reference
+                              | expression_pipeline_reference
 
             // reference rules:
-            expression_pipeline_reference: "pipeline" "()" "." EXPRESSION_PIPELINE_PROPERTY "." EXPRESSION_PARAMETER_NAME
             expression_variable_reference: "variables" "(" EXPRESSION_VARIABLE_NAME ")"
-            expression_activity_reference: "activity" "(" EXPRESSION_ACTIVITY_NAME ")"
-            expression_dataset_reference: "dataset" "()" "." EXPRESSION_PARAMETER_NAME
-            expression_linked_service_reference: "linkedService" "()" "." EXPRESSION_PARAMETER_NAME
+            expression_datafactory_parameters_reference: EXPRESSION_DATAFACTORY_REFERENCE "()"
+            expression_datafactory_activity_reference: "activity" "(" EXPRESSION_ACTIVITY_NAME ")"
             expression_item_reference: "item" "()"
-            expression_system_variable_reference: "pipeline" "()" "." EXPRESSION_SYSTEM_VARIABLE_NAME
+            expression_parameter_reference: "parameter" "(" EXPRESSION_PARAMETER_NAME ")"
```

Comment:
Is this expression still used? It's not data factory expression language afaik and if it's logic apps expression language I don't see any transformations using this. Curious!